sawmill-benchmark/src/main/java/io/logz/sawmill/benchmark/SawmillMicroBenchmark.java

```diff
@@ -0,0 +1,217 @@
+                IntStream.range(from, from + howmany).boxed().collect(Collectors.toMap(
+                        i -> templateService.createTemplate("FieldFrom" + i),
+                        i -> templateService.createTemplate("FieldTo" + i)
+                ))
+        );
+    }
+
+    @Benchmark
+    public void benchmarkPipelineWithOneRename() {
+        Doc doc = createDocFromTemplate(documentTemplate);
+        ExecutionResult executionResult = pipelineExecutor.execute(pipelineWithOneRename, doc);
+    }
+
+    @Benchmark
+    public void benchmarkPipelineWithOneRemove() {
+        Doc doc = createDocFromTemplate(documentTemplate);
+        ExecutionResult executionResult = pipelineExecutor.execute(pipelineWithOneRemoveProcessor, doc);
+    }
+
+    @Benchmark
+    public void benchmarkPipelineWithTwoRenameProcessors() {
+        Doc doc = createDocFromTemplate(documentTemplate);
+        ExecutionResult executionResult = pipelineExecutor.execute(pipelineWithTwoRenameProcessors, doc);
+    }
+
+    @Benchmark
+    public void benchmarkPipelineWithTwoRenamesInOneProcessor() {
+        Doc doc = createDocFromTemplate(documentTemplate);
+        ExecutionResult executionResult = pipelineExecutor.execute(pipelineWithTwoRenamesInOneProcessor, doc);
+    }
+
+    @Benchmark
+    public void benchmarkPipelineWithVariableRenamesInOneProcessor() {
+        Doc doc = createDocFromTemplate(documentTemplate);
+        ExecutionResult executionResult = pipelineExecutor.execute(pipelineWithVariableRenamesInOneProcessor, doc);
+    }
+
+    @Benchmark
+    public void benchmarkPipelineWithVariableRenameProcessors() {
+        Doc doc = createDocFromTemplate(documentTemplate);
+        ExecutionResult executionResult = pipelineExecutor.execute(pipelineWithVariableRenameProcessors, doc);
+    }
+
+    @Benchmark
+    public void benchmarkPipelineWithVariableRemoveProcessors() {
+        Doc doc = createDocFromTemplate(documentTemplate);
+        ExecutionResult executionResult = pipelineExecutor.execute(pipelineWithVariableRemoveProcessors, doc);
+    }
+
+    private Pipeline createPipeline(ExecutionStep... steps) {
+        return createPipeline(false, steps);
+    }
+
+    private Pipeline createPipeline(boolean stopOnFailure, ExecutionStep... steps) {
+        String id = "abc";
+        return new Pipeline(id, Arrays.asList(steps), stopOnFailure);
+    }
+
+    public Doc createDoc(Object... objects) {
+        if (objects.length % 2 != 0) {
+            throw new RuntimeException("Can't construct map out of uneven number of elements");
+        }
+
+        LinkedHashMap map = new LinkedHashMap<>();
+        if (objects != null) {
+            for (int i = 0; i < objects.length; i++) {
+                map.put(objects[i], objects[++i]);
+            }
+        }
+
+        return new Doc(map);
+    }
+
+    public Doc createDocFromTemplate(Map<String, String> original) {
+        return new Doc(new LinkedHashMap<>(original));
+    }
```

Comment:
```suggestion
    public Doc createDocFromTemplate(Map<String, Object> original) {
        return new Doc(original);
    }
```

Reply:
Pipeline modified the original map. Have to create a copy here.

Reply:
🤷